src/components/app_catalog/app_list/index.js

```diff
@@ -43,26 +59,23 @@ class AppList extends React.Component {
     };
   }
 
-  // filterApps is a pure function that takes a filter a list of all apps and returns
+  // filterApps is a function that takes a filter and a list of all apps and returns
   // a filtered set of apps
   filterApps(allApps, filter) {
-    let apps = Array.from(Object.entries(allApps));
-    apps = apps.map(([, app]) => {
-      app.name = app[0].name;
-      app.version = app[0].version;
-      app.icon = app[0].icon;
-      return app;
-    });
+    // Lunr search
+    var lunrResults = this.index
+      .search(filter.searchQuery.trim() + ' ' + filter.searchQuery.trim() + '*')
```

Comment:
I dont get why it is repeated, can you explain to me?

Reply:
Yeah, it has to do with how lunr handles the wildcard (`*`)

Without the double query like it is now, if we search for `prom` then the query will become `prom*`. And that will find `prometheus`.

However If we search for `prometheus` then the query will become `prometheus*` which will not match for `prometheus` since it is looking for something that has 1 or more characters after the string `prometheus`

So this is the workaround to allow the  'search as you type' behaviour we had before, without it  breaking as soon as you type the full matching word.

Reply:
ah ok so you can not use something like '+'